### PR TITLE
Made some functions in time module const

### PR DESCRIPTION
**Pull request by rizakrko**
They may be const, or i missed something?

### CODE REVIEW

File: src/libcore/time.rs [hunk 1]
```diff
@@ -204,7 +204,7 @@ impl Duration {
     /// [`subsec_nanos`]: #method.subsec_nanos
     #[stable(feature = "duration", since = "1.3.0")]
     #[inline]
-    pub fn as_secs(&self) -> u64 { self.secs }
+    pub const fn as_secs(&self) -> u64 { self.secs }
```
Comment (oli-obk) [inline src/libcore/time.rs]: can you add `#[rustc_const_unstable="duration_getters"]` to the newly const-ed functions so they aren't immediately constant on the next stable rust?

**Review by oli-obk — CHANGES_REQUESTED** [review]
I think it would be good to have some tests for these functions showing various conversions between different duration units

**Comment by oli-obk** [comment]
Oh... I should have explained better. Could you add some tests that do this kind of conversion at compile-time?

So e.g. use it for computing an array length or something weird like that.

**Comment by rizakrko** [comment]
That makes sense. I'l try to come up with something.Also, where should I put these tests?

**Comment by oli-obk** [comment]
> That makes sense. I'l try to come up with something.Also, where should I put these tests?

I usually throw them into https://github.com/rust-lang/rust/tree/master/src/test/ui/const-eval as a `// compile-pass` test (https://github.com/rust-lang/rust/blob/master/src/test/ui/const-eval/const_signed_pat.rs is an example)

File: src/test/ui/const-eval/duration_conversion.rs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// compile-pass
```
[the remark this answers is not in the document]
Comment (rizakrko) [reply src/test/ui/const-eval/duration_conversion.rs]: Sorry for that, it should not pass. I will update PR

**Review by oli-obk — APPROVED** [review]

**Comment by oli-obk** [comment]
OK Lgtm now, please squash the commits down to one, then we can send it off to bors

**Comment by oli-obk** [comment]
@bors rollup

This is not gonna conflict with anything

**Pull request merged**